libsolidity/formal/BMC.cpp

```diff
@@ -650,7 +654,12 @@ pair<vector<smtutil::Expression>, vector<string>> BMC::modelExpressions()
 		if (uf->annotation().type->isValueType())
 		{
 			expressionsToEvaluate.emplace_back(expr(*uf));
-			expressionNames.push_back(uf->location().text());
+			string expressionName;
+			if (uf->location().hasText())
+				expressionName = m_charStreamProvider.charStream(*uf->location().sourceName).text(
+					uf->location()
+				);
```

Comment:
```suggestion
			expressionName = m_charStreamProvider.charStream(*uf->location().sourceName).text(
				uf->location()
			);
```

Reply:
argh, right

Reply:
actually to be honest: We can still have a nullptr in `sourceName`, so we have to check at least for that.

Reply:
Doesn't the `hasText()` check in `text()` cover tha already?

Reply:
ok, I see it's about the `*uf->location().sourceName` part. Fine then.